compiler/packages/react-compiler-runtime/src/index.ts

```diff
@@ -19,22 +19,27 @@ const ReactSecretInternals =
 type MemoCache = Array<number | typeof $empty>;
 
 const $empty = Symbol.for('react.memo_cache_sentinel');
-/**
- * DANGER: this hook is NEVER meant to be called directly!
- **/
-export function c(size: number) {
-  return React.useState(() => {
-    const $ = new Array(size);
-    for (let ii = 0; ii < size; ii++) {
-      $[ii] = $empty;
-    }
-    // This symbol is added to tell the react devtools that this array is from
-    // useMemoCache.
-    // @ts-ignore
-    $[$empty] = true;
-    return $;
-  })[0];
-}
+
+// Re-export React.c if present, otherwise fallback to the userspace polyfill for versions of React
+// < 19.
+export const c =
+  // @ts-expect-error
+  typeof React.c === 'function'
+    ? // @ts-expect-error
+      React.c
+    : function c(size: number) {
+        return React.useState<Array<unknown>>(() => {
```

Comment:
aside: does `useState` reset when a fast refresh occurs that affects the given component? if not we could maybe use useMemo w no deps instead

Reply:
From what I can see it preserves state on FastRefresh, it only resets if the state variable got renamed